scripts/e2e-cluster-dump.sh

```diff
@@ -0,0 +1,182 @@
+    for pod in $mayastor_csipods
+    do
+        echo "# $pod csi-driver-registrar $* ---------------------------------"
+        kubectl -n mayastor logs "$@" "$pod"  csi-driver-registrar
+    done
+
+    moacpod=$(kubectl -n mayastor get pods | grep moac | sed -e 's/ .*//')
+    echo "# $moacpod csi-provisioner $* ---------------------------------"
+    kubectl -n mayastor logs "$@" "$moacpod" csi-provisioner
+    echo "# $moacpod csi-attacher $* ---------------------------------"
+    kubectl -n mayastor logs "$@" "$moacpod" csi-attacher
+}
+
+function logs-csi-mayastor {
+    mayastor_csipods=$(kubectl -n mayastor get pods | grep mayastor-csi | sed -e 's/ .*//')
+    for pod in $mayastor_csipods
+    do
+        echo "# $pod mayastor-csi $* ---------------------------------"
+        kubectl -n mayastor logs "$@" "$pod"  mayastor-csi
+    done
+}
+
+function logs-mayastor {
+    mayastor_pods=$(kubectl -n mayastor get pods | grep mayastor | grep -v mayastor-csi | sed -e 's/ .*//')
+    for pod in $mayastor_pods
+    do
+        echo "# $pod mayastor $* ---------------------------------"
+        kubectl -n mayastor logs "$@" "$pod"  mayastor
+    done
+}
+
+function logs-moac {
+    moacpod=$(kubectl -n mayastor get pods | grep moac | sed -e 's/ .*//')
+    echo "# $moacpod moac $* ---------------------------------"
+    kubectl -n mayastor logs "$@" "$moacpod" moac
+}
+
+# $1 = podlogs, 0 => do not generate pod logs
+function dump-to-stdout {
+    if [ "$1" -ne 0 ]; then
+        logs-moac
+        logs-mayastor
+        logs-csi-mayastor
+        logs-csi-containers
+
+        logs-moac -p
+        logs-mayastor -p
+        logs-csi-mayastor -p
+        logs-csi-containers -p
+    fi
+
+    cluster-get
+    cluster-describe
+}
+
+# $1 = podlogs, 0 => do not generate pod logs
+# $2 = dest  mkdir $dest and generate logs there.
+function dump-to-dir {
+    dest="$2"
+    echo "Generating logs in $dest"
+    mkdir -p "$dest"
+    if [ "$1" -ne 0 ]; then
+        logs-moac >& "$dest/moac.log"
+        logs-mayastor >& "$dest/mayastor.log"
+        logs-csi-mayastor >& "$dest/csi-mayastor.log"
+        logs-csi-containers >& "$dest/csi-containers.log"
+
+        logs-moac -p >& "$dest/moac.previous.log"
+        logs-mayastor -p >& "$dest/mayastor.previous.log"
+        logs-csi-mayastor -p >& "$dest/csi-mayastor.previous.log"
+        logs-csi-containers -p >& "$dest/csi-containers.previous.log"
+    fi
+
+    cluster-get >& "$dest/cluster.get.txt"
+    cluster-describe >& "$dest/cluster.describe.txt"
+}
+
+# $1 = podlogs, 0 => do not generate pod logs
+# $2 = [destdir] undefined => dump to stdout,
+#                   otherwise generate log files in $destdir
+function dump {
+    if [ -z "$2" ]; then
+        dump-to-stdout "$1"
+    else
+        dump-to-dir "$1" "$2"
+    fi
+}
+
+podlogs=1
+destdir=
+
+# Parse arguments
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+    -p|--path)
```

Comment:
For the sake of consistency across layers. Eases up orientation when reading if the same thing has the same name.
```suggestion
    -d|--destdir)
```

Reply:
Done